src/server/network/network.ts

```diff
@@ -0,0 +1,93 @@
+import dgram = require('dgram');
+
+export type ClientListener = (event: string, ...args: any[]) => void;
+
+type TeamData = {
+  server: dgram.Socket;
+  name: string;
+  address: string;
+  port: number;
+};
+
+export class UDPServer {
+  private redTeam: TeamData;
+  private blueTeam: TeamData;
+  private clients: Map<string, ClientListener>;
+
+  constructor(redTeamAddress: string, blueTeamAddress: string) {
+    this.clients = new Map();
+
+    this.redTeam = {
+      server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
```

Comment:
I'd take the effort to make it support ipv6 if it's not too hard now rather than do it later